src/SFML/Graphics/RenderTarget.cpp

```diff
@@ -317,6 +319,107 @@ void RenderTarget::draw(const Vertex* vertices, std::size_t vertexCount,
+        return;
+    }
+
+    // Sanity check
+    if (firstVertex > vertexBuffer.m_size)
+        return;
+
+    // Clamp vertexCount to something that makes sense
+    vertexCount = std::min(vertexCount, vertexBuffer.m_size - firstVertex);
+
+    // Nothing to draw?
+    if (!vertexCount || !vertexBuffer.m_buffer)
+        return;
+
+    // GL_QUADS is unavailable on OpenGL ES
+    #ifdef SFML_OPENGL_ES
+        if (vertexBuffer.m_primitiveType == Quads)
+        {
+            err() << "sf::Quads primitive type is not supported on OpenGL ES platforms, drawing skipped" << std::endl;
+            return;
+        }
+        #define GL_QUADS 0
+    #endif
+
+    if (setActive(true))
+    {
+        // First set the persistent OpenGL states if it's the very first call
+        if (!m_cache.glStatesSet)
+            resetGLStates();
+
+        applyTransform(states.transform);
+
+        // Apply the view
+        if (m_cache.viewChanged)
+            applyCurrentView();
+
+        // Apply the blend mode
+        if (states.blendMode != m_cache.lastBlendMode)
+            applyBlendMode(states.blendMode);
+
+        // Apply the texture
+        Uint64 textureId = states.texture ? states.texture->m_cacheId : 0;
+        if (textureId != m_cache.lastTextureId)
+            applyTexture(states.texture);
+
+        // Apply the shader
+        if (states.shader)
+            applyShader(states.shader);
+
+        // Bind vertex buffer
+        VertexBuffer::bind(&vertexBuffer);
+
+        // Always enable texture coordinates
+        if (!m_cache.texCoordsArrayEnabled)
+            glCheck(glEnableClientState(GL_TEXTURE_COORD_ARRAY));
+
+        glCheck(glVertexPointer(2, GL_FLOAT, sizeof(Vertex), reinterpret_cast<const void*>(0)));
+        glCheck(glColorPointer(4, GL_UNSIGNED_BYTE, sizeof(Vertex), reinterpret_cast<const void*>(8)));
+        glCheck(glTexCoordPointer(2, GL_FLOAT, sizeof(Vertex), reinterpret_cast<const void*>(12)));
+
+        // Find the OpenGL primitive type
+        static const GLenum modes[] = {GL_POINTS, GL_LINES, GL_LINE_STRIP, GL_TRIANGLES,
+                                       GL_TRIANGLE_STRIP, GL_TRIANGLE_FAN, GL_QUADS};
+        GLenum mode = modes[vertexBuffer.m_primitiveType];
+
+        // Draw the primitives
+        glCheck(glDrawArrays(mode, firstVertex, vertexCount));
+
+        // Unbind vertex buffer
+        VertexBuffer::bind(NULL);
+
+        // Unbind the shader, if any
+        if (states.shader)
+            applyShader(NULL);
+
+        // If the texture we used to draw belonged to a RenderTexture, then forcibly unbind that texture.
+        // This prevents a bug where some drivers do not clear RenderTextures properly.
+        if (states.texture && states.texture->m_fboAttachment)
+            applyTexture(NULL);
+
+        // Update the cache
+        m_cache.useVertexCache = false;
+        m_cache.texCoordsArrayEnabled = true;
+    }
+}
```

Comment:
I didn't check in detail, but maybe we could refactor some stuff, common between VA and VB drawing? At least the GL -> SFML primitive type conversion 😸

Reply:
Maybe...